### PR TITLE
MAUI eBook configuration management

### PR DESCRIPTION
**Pull request by michaelstonis**
## Summary

Configuration management chapter for .NET MAUI eBook

### CODE REVIEW

**Review by IEvangelist — COMMENTED** [review]
A few minor nits

**Comment by IEvangelist** [comment]
These are the warnings that are causing build errors (that will need to be addressed before approving and merging):

The _docs/architecture/maui/index.md_ file:

- Line 73: [Warning] Link 'https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app' contains locale code 'en-us'. For localizability, remove 'en-us' from links to most Microsoft sites.
- Line 77: [Warning] Link 'https://docs.microsoft.com/en-us/dotnet/maui/' contains locale code 'en-us'. For localizability, remove 'en-us' from links to most Microsoft sites.
- Line 84: [Warning] Invalid file link: 'introduction.md'.
- Line 73: [Suggestion] Absolute link 'https://docs.microsoft.com/en-us/dotnet/maui/get-started/first-app' will be broken in isolated environments. Replace with a relative link.
- Line 77: [Suggestion] Absolute link 'https://docs.microsoft.com/en-us/dotnet/maui/' will be broken in isolated environments. Replace with a relative link.

I'd recommend removing the locales from all URLs, they're not needed and cause warnings/errors. Instead of absolute links, use relative.

The _docs/architecture/maui/toc.yml_ file:

- Line 5: [Warning] Invalid file link: 'preface.md'.
- Line 7: [Warning] Invalid file link: 'introduction.md'.
- Line 9: [Warning] Invalid file link: 'mvvm.md'.
- Line 11: [Warning] Invalid file link: 'dependency-injection.md'.
- Line 13: [Warning] Invalid file link: 'communicating-between-components.md'.
- Line 15: [Warning] Invalid file link: 'navigation.md'.
- Line 17: [Warning] Invalid file link: 'validation.md'.
- Line 21: [Warning] Invalid file link: 'micro-services.md'.
- Line 23: [Warning] Invalid file link: 'authentication-and-authorization.md'.
- Line 25: [Warning] Invalid file link: 'unit-testing.md'.
- Line 27: [Warning] Invalid file link: 'accessing-remote-data.md'.

I'd recommend adding these files, even if they're empty. This will remove the warnings/errors, and allow you to have a solid outline for the eBook.

File: docs/architecture/maui/configuration-management.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    public string AuthAccessToken
+    {
+        get => Preferences.Get(AccessToken, AccessTokenDefault);
+        set => Preferences.Set(AccessToken, value);
+    }
+
+    public bool UseMocks
+    {
+        get => Preferences.Get(IdUseMocks, UseMocksDefault);
```
[the remark this answers is not in the document]
Comment (michaelstonis) [reply docs/architecture/maui/configuration-management.md:64]: There was a section above that felt out of place, but provided this context. I moved it to the top of the section setup and I think it reads better and provides a bit more situational context.

File: docs/architecture/maui/configuration-management.md [hunk 2]
```diff
@@ -1,9 +1,133 @@
...
+
+.NET MAUI includes a preferences manager that provides a way to store runtime settings for a user. This feature can be accessed from anywhere within your application using the `Microsoft.Maui.Storage.Preferences` class. The preferences manager provides a consistent, type-safe, cross-platform approach for persisting and retrieving app and user settings, while using the native settings management provided by each platform. In addition, it's straightforward to use data binding to access settings data exposed by the library. For more information, see the [Preferences](/dotnet/maui/platform-integration/storage/preferences) on the Microsoft Developer Center.
+
+> [!TIP]
+> Preferences is meant for storing relatively small data. If you need to store larger or more complex data, consider using a local database or filesystem to store the data.
+
+Our application will use the `Preferences` class need to implement the `ISettingsService` interface. The code below shows how the eShopOnContainers mobile app's `SettingsService` implements the `AuthTokenAccess` and `UseMocks` properties:
+
+```csharp
+public class SettingsService : ISettingsService
```
Comment (IEvangelist) [inline docs/architecture/maui/configuration-management.md:48]:
```suggestion
public sealed class SettingsService : ISettingsService
```

File: docs/architecture/maui/configuration-management.md [hunk 3]
```diff
@@ -1,9 +1,133 @@
...
+}
+
+public string IdentityEndpoint
+{
+    get => _identityEndpoint;
+    set
+    {
+        SetProperty(ref _identityEndpoint, value);
+
+        if (!string.IsNullOrEmpty(value))
```
Comment (IEvangelist) [inline docs/architecture/maui/configuration-management.md:110]: This code tells me, the reader, that whitespace is a valid value. If it is not, please consider using `string.IsNullOrWhitespace` instead.

**Review by IEvangelist — COMMENTED** [review]
Almost there.

File: docs/architecture/maui/configuration-management.md (hunk 3 above)
Comment (IEvangelist) [inline docs/architecture/maui/configuration-management.md:110]:
```suggestion
        if (!string.IsNullOrWhitespace(value))
```

**Pull request merged**